src/blas/backends/rocblas/rocblas_level2.cpp

```diff
@@ -0,0 +1,2736 @@
+                    dependencies);                                                              \
+    }
+
+GEMV_LAUNCHER_USM(float, rocblas_sgemv)
+GEMV_LAUNCHER_USM(double, rocblas_dgemv)
+GEMV_LAUNCHER_USM(std::complex<float>, rocblas_cgemv)
+GEMV_LAUNCHER_USM(std::complex<double>, rocblas_zgemv)
+#undef GEMV_LAUNCHER_USM
+
+template <typename Func, typename T>
+inline cl::sycl::event gbmv(Func func, cl::sycl::queue &queue, transpose trans, int64_t m,
+                            int64_t n, int64_t kl, int64_t ku, T alpha, const T *a, int64_t lda,
+                            const T *x, int64_t incx, T beta, T *y, int64_t incy,
+                            const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
+    using rocDataType = typename RocEquivalentType<T>::Type;
+    overflow_check(n, m, lda, kl, ku, incx, incy);
+    auto done = queue.submit([&](cl::sycl::handler &cgh) {
+        int64_t num_events = dependencies.size();
+        for (int64_t i = 0; i < num_events; i++) {
+            cgh.depends_on(dependencies[i]);
+        }
+        onemkl_rocblas_host_task(cgh, queue, [=](RocblasScopedContextHandler &sc) {
+            auto handle = sc.get_handle(queue);
+
+            auto a_ = reinterpret_cast<const rocDataType *>(a);
+            auto x_ = reinterpret_cast<const rocDataType *>(x);
+            auto y_ = reinterpret_cast<rocDataType *>(y);
+            rocblas_status err;
+            ROCBLAS_ERROR_FUNC(func, err, handle, get_rocblas_operation(trans), m, n, kl, ku,
+                               (rocDataType *)&alpha, a_, lda, x_, incx, (rocDataType *)&beta, y_,
+                               incy);
+        });
+    });
+    return done;
+}
+
+#define GBMV_LAUNCHER_USM(TYPE, ROCBLAS_ROUTINE)                                                  \
+    cl::sycl::event gbmv(cl::sycl::queue &queue, transpose trans, int64_t m, int64_t n,           \
+                         int64_t kl, int64_t ku, TYPE alpha, const TYPE *a, int64_t lda,          \
+                         const TYPE *x, int64_t incx, TYPE beta, TYPE *y, int64_t incy,           \
+                         const cl::sycl::vector_class<cl::sycl::event> &dependencies) {           \
+        return gbmv(ROCBLAS_ROUTINE, queue, trans, m, n, kl, ku, alpha, a, lda, x, incx, beta, y, \
+                    incy, dependencies);                                                          \
+    }
+
+GBMV_LAUNCHER_USM(float, rocblas_sgbmv)
+GBMV_LAUNCHER_USM(double, rocblas_dgbmv)
+GBMV_LAUNCHER_USM(std::complex<float>, rocblas_cgbmv)
+GBMV_LAUNCHER_USM(std::complex<double>, rocblas_zgbmv)
+#undef GBMV_LAUNCHER_USM
+
+template <typename Func, typename T>
+inline cl::sycl::event ger(Func func, cl::sycl::queue &queue, int64_t m, int64_t n, T alpha,
+                           const T *x, int64_t incx, const T *y, int64_t incy, T *a, int64_t lda,
+                           const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
```

Comment:
Here and other locations.
```suggestion
                           const std::vector<cl::sycl::event> &dependencies) {
```